src/utils/errors.js

```diff
@@ -7,57 +7,91 @@
+var messages = {
+  warn: {
+    childViewWasPassedAsObjectWithoutAScopeProperty: () => 'ChildView was passed as object without a scope property',
+    collectionMethodMayNotBeOverridden: function(methodName, debugName) {
+      return `Collection.${methodName} may not be overridden` + (debugName ? ` for collection "${debugName}"` : '');
+    },
+    collectionExpectedArrayOfObjectsButGot: (initialStr) => `Collection expected array of objects but got: ${initialStr}`,
+    modelMethodMayNotBeOverridden: function(methodName, debugName) {
+      return `Model.${methodName} may not be overridden` + (debugName ? ` for model "${debugName}"` : '');
+    },
+    modelExpectedObjectOfAttributesButGot: (initialStr) => `Model expected object of attributes but got: ${initialStr}`,
+    domDoesNotMatchVDOMForView: (debugName) => `DOM does not match VDOM for view "${debugName}". Use debug panel to see differences`,
+    viewMethodMayNotBeOverridden: function(methodName, debugName) {
+      return `View.${methodName} may not be overridden` + (debugName ? ` for view "${debugName}"` : '');
+    },
+    componentFunctionMayNotBeCalledDirectly: (fn) => `Component.${fn} may not be called directly`,
+    cannotOverwriteComponentMethod: (fn) => `Cannot overwrite component method: ${fn}`,
+    unableToParseToAValidValueMustMatchJSONFormat: (value) => `Unable to parse "${value}" to a valid value. Input must match JSON format`,
+    widgetTypeHasNoTemplateToStringFunctionFallingBackToDOM: (name) => `Widget type: ${name} has no templateToString function, falling back to DOM`,
+    objectDoesNotMeetExpectedEventSpec: () => 'Object does not meet expected event spec',
+    warningNoPartialRegisteredWithTheName: (partialName) => `Warning: no partial registered with the name ${partialName}`,
+    doubleCurlyInterpolatorsCannotBeInAttributes: () => 'Double curly interpolators cannot be in attributes',
+    tagsImproperlyPairedClosing: (tagName, openID, id) => `${tagName} tags improperly paired, closing ${openID} with close tag from ${id}`,
+    closingElementWhenTheStackWasEmpty: (id) => `Closing element ${id} when the stack was empty`,
+    computedPropertiesAreNowDeprecatedPleaseChange: (name) => `Computed properties are now deprecated and will be removed soon. Please change "${name}" to a derived property`
+  },
+  error: {
+    unableToLaunchDebugPanel: () => 'Unable to launch debug panel. You may need to allow the popup or run "window.launchDebugger()" from your console'
+  },
+  exception: {
+  }
+};
+
+/**
+ * Returns a logger call with the provided message and type
+ * @param  {function} message  function that returns our message
+ * @param  {string} type The type of log to make (warning, error, exception, etc...)
+ * @return {function}      Function that calls logger with the provided message and type
+ */
+function loggerize(message, type) {
+  return function() {
+    logger[type](message.apply(message, arguments));
```

Comment:
to allow multiple arguments to get passed through, you could do
```
var output = message.apply(message, arguments);
if (_.isArray(output)) {
  logger[type].apply(logger, output);
} else {
  logger[type](output);
}
return output;
```

that would let you do things like:
```
objectDoesNotMeetExpectedEventSpec: (evt) => ['Object does not meet expected event spec', evt]
```